server/src/services/Google.ts

```diff
@@ -69,7 +70,9 @@ function parseEventData({
     ...(summary && { summary }),
     // Since Google will send emails to these addresses, we don't want to
     // accidentally send emails in testing.
-    ...(attendees && { attendees: isProd() || isTest() ? attendees : [] }),
+    // Additionally include attendees only for future events.
+    ...((!start || isFuture(start)) &&
+      attendees && { attendees: isProd() || isTest() ? attendees : [] }),
```

Comment:
```suggestion
    ...(attendees && { attendees: isProd() || isTest() ? attendees : [] }),
```
Unfortunately this doesn't quite work how you'd expect.  If you don't provide a list of attendees the calendar api will remove any existing attendees.

The approach has to be to send whatever you want the event to become after the call.  In this case, if it's an event update we just want to reflect back the existing attendee list.

For event creation, I think we just have to handle that as a special case.

Reply:
Hmm, I believe that's what's happening already. There's test for updating past event and ignoring changed attendees in the send `eventData` - https://github.com/gikf/chapter/blob/d73cb953ff5ad572bd58e2964b1a1d40e83eaca2/server/tests/google.test.ts#L316
I've locally added another one, updating future event, and it also kept attendees intact.

If commented code won't include `attendees` in the parsed data, which means that result of `createEventRequestBody` call won't have it, the original attendees from `data` will be used.

https://github.com/freeCodeCamp/chapter/blob/8d6d0145cfd24c6b3c5037d87bd36ecd97a92432/server/src/services/Google.ts#L140-L147

Or is there some another edge case?

Reply:
Good point. No, I don't believe there is another edge case.

While I was digging into this, I did some pretty heavy refactoring.  Do you mind if I push a commit?

Reply:
Cheers, I think this is a little better.

Reply:
I agree, I like it.